### PR TITLE
Feature/delivery mode support: make Semester 2 Active

### PR DESCRIPTION
**Pull request by manu354**
1. Make sem2 active now, since almost everyone looking around this time would be looking to generate their sem2 timetable 

2. Change n locations to "On Campus". I thought having the location text everywhere is a bit confusing as it is not clear that it means there is 1 on campus location

### CODE REVIEW

**Comment by Trontor** [comment]
> Change n locations to "On Campus". I thought having the location text everywhere is a bit confusing as it is not clear that it means there is 1 on campus location

This makes the assumption that every class with 1 location is "On Campus". Is there a possibility for there to be one location, that is, some variant of "Online"/"Zoom"? Would be great to have confidence that we aren't misleading people here.

File: client/src/utility/ClassRender.js [hunk 1]
```diff
@@ -35,10 +35,18 @@ export default function ({event, el}) {
 
   if (height >= CLASS_LOCATION_MINIMUM.height && width >= CLASS_LOCATION_MINIMUM.width) {
     let locationsText;
-    if(!online){
+    if(!online && locations.length> 1){
```
Comment (Trontor) [inline client/src/utility/ClassRender.js:38]: I'd like more confidence that this invariant holds as you assume it does.

**Comment by manu354** [comment]
Hey this is the code that parses class data to set the location to either 'online' or the actual location. 

I can add .includes("zoom") to this and other keywords that would suggest it is online, but all classes I have seen either have no location, or mention being online. 

`const parseLocation = (location: string) =>{
  // Because an empty string, whitespace, or a string that contains the word online indicates that
  // the subject is online
  const isOnline = location.toLowerCase().includes('online') || location == '' || !location.trim();
  return isOnline ? 'online' :  location;
}`

File: client/src/utility/ClassRender.js (hunk 1 above)
Comment (Trontor) [inline client/src/utility/ClassRender.js:38]: nite: formatting

File: client/src/utility/ClassRender.js [hunk 2]
```diff
@@ -35,10 +35,18 @@ export default function ({event, el}) {
...
           `<div class="fc-loc">
-        ${locations.length} location${locations.length > 1 ? 's' : ''}
+          
+         ${locations.length} locations
+      </div>`
+      );
+    }
+    else if(!online) {
+      locationsText = $(
+        `<div class="fc-loc">
```
Comment (Trontor) [inline client/src/utility/ClassRender.js:48]: nit: formatting (open tag not aligned with close)

**Comment by Trontor** [comment]
> Hey this is the code that parses class data to set the location to either 'online' or the actual location.
> 
> I can add .includes("zoom") to this and other keywords that would suggest it is online, but all classes I have seen either have no location, or mention being online.
> 
> ```
> const parseLocation = (location: string) =>{
>   // Because an empty string, whitespace, or a string that contains the word online indicates that
>   // the subject is online
>   const isOnline = location.toLowerCase().includes('online') || location == '' || !location.trim();
>   return isOnline ? 'online' :  location;
> }
> ```
> 
> based on what this function returns, the front-end will determine whether the subject is in-person or online, if it is online the locations length will be 0.

Great, looks good to me! Just some formatting touch ups and it's good to merge.

**Comment by manu354** [comment]
Fixed formatting and will have prettier run automatically from now on :)

**Pull request merged**